arch/x86/coco/tdx/tdx.c

```diff
@@ -276,6 +276,20 @@ static void enable_cpu_topology_enumeration(void)
 	tdg_vm_wr(TDCS_TD_CTLS, TD_CTLS_ENUM_TOPOLOGY, TD_CTLS_ENUM_TOPOLOGY);
 }
 
+static void reduce_unnecessary_ve(void)
+{
+	u64 err = tdg_vm_wr(TDCS_TD_CTLS, TD_CTLS_REDUCE_VE, TD_CTLS_REDUCE_VE);
```

Comment:
This cannot be set if x2APIC_ID has not been properly configured yet. Need to read TDCS_TOPOLOGY_ENUM_CONFIGURED to check that. The other commit does this. Do the same here.